api/v1beta1/proposal_webhook.go

```diff
@@ -95,6 +95,12 @@ func (r *Proposal) ValidateCreate(ctx context.Context, client client.Client, use
 		return err
 	}
 
+	if r.Spec.UpgradeChaincode != nil {
+		if err := validateChaincodePhase(ctx, client, r.Spec.UpgradeChaincode.Chaincode); err != nil {
+			return err
+		}
+	}
+
 	return validateChaincodeBuild(ctx, client, r.Spec.ProposalSource)
```

Comment:
Also this `validateChaincodeBuild` seems belongs to validate logic when proposal is `deploychaincode/upgradechaincode`

Reply:
@0xff-dev how about this one?